src/overlays/actors/ovl_En_Ge3/z_en_ge3.c

```diff
@@ -28,49 +46,386 @@ const ActorInit En_Ge3_InitVars = {
+                case GERUDO_AVEIL_CSACTION_BEEHIVE_RUN_AWAY:
+                    EnGe3_ChangeAnim(this, GERUDO_AVEIL_ANIM_RUN_AWAY, ANIMMODE_LOOP, 0.0f);
+                    this->picto.actor.speedXZ = 5.0f;
+                    this->screamTimer = (s32)(Rand_ZeroFloat(10.0f) + 20.0f);
+                    break;
+
+                case GERUDO_AVEIL_CSACTION_BEEHIVE_VANISH:
+                    Actor_MarkForDeath(&this->picto.actor);
+                    break;
+            }
+
+            SkelAnime_Update(&this->skelAnime);
+        }
+    } else if ((this->picto.actor.xzDistToPlayer < 700.0f) && (fabsf(this->picto.actor.playerHeightRel) < 100.0f) &&
+               !Play_InCsMode(play)) {
+        func_800B7298(play, &this->picto.actor, 0x1A);
+        func_801000A4(NA_SE_SY_FOUND);
+
+        if (Player_GetMask(play) == PLAYER_MASK_STONE) { // Not fooled by Stone Mask
+            Message_StartTextbox(play, 0x11AF, &this->picto.actor);
+        } else {
+            Message_StartTextbox(play, 0x11AE, &this->picto.actor);
+        }
+
+        this->actionFunc = EnGe3_ThrowPlayerOut;
+        this->actionTimer = 50;
+        gSaveContext.save.weekEventReg[80] |= 8; // Aveil discovered Player
+    }
+
+    if (this->csAction == GERUDO_AVEIL_CSACTION_BEEHIVE_RUN_AWAY) {
+        this->picto.actor.speedXZ = 5.0f;
+        EnGe3_FollowPath(this);
+
+        if (Animation_OnFrame(&this->skelAnime, 2.0f) || Animation_OnFrame(&this->skelAnime, 6.0f)) {
+            Actor_PlaySfxAtPos(&this->picto.actor, NA_SE_EV_PIRATE_WALK);
+        }
+
+        if (this->screamTimer > 0) {
+            this->screamTimer--;
+        } else {
+            this->screamTimer = (s32)(Rand_ZeroFloat(10.0f) + 20.0f);
+            EnGe3_Scream(this);
+        }
+    }
+}
+
+void EnGe3_UpdateColliderAndMove(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnGe3* this = (EnGe3*)thisx;
+
+    Collider_UpdateCylinder(&this->picto.actor, &this->collider);
+    CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+    Actor_MoveWithGravity(&this->picto.actor);
+    Actor_UpdateBgCheckInfo(play, &this->picto.actor, 40.0f, 25.0f, 40.0f, 4);
+}
+
+void EnGe3_Blink(EnGe3* this, PlayState* play) {
+    if (DECR(this->blinkTimer) == 0) {
+        this->blinkTimer = Rand_S16Offset(60, 60);
+    }
+
+    this->eyeIndex = this->blinkTimer;
+
+    if (this->eyeIndex >= 3) {
+        this->eyeIndex = 0;
+    }
+}
+
+void EnGe3_Update(Actor* thisx, PlayState* play) {
+    EnGe3* this = (EnGe3*)thisx;
 
-extern ColliderCylinderInit D_809A0DA0;
+    EnGe3_UpdateColliderAndMove(&this->picto.actor, play);
+    this->actionFunc(this, play);
+    EnGe3_Blink(this, play);
+}
 
-extern UNK_TYPE D_0600A808;
+s32 EnGe3_ValidatePictograph(PlayState* play, Actor* thisx) {
+    s32 ret = Snap_ValidatePictograph(play, thisx, PICTOGRAPH_PIRATE_GOOD, &thisx->focus.pos, &thisx->shape.rot, 10.0f,
+                                      400.0f, -1);
```

Comment:
Is it even possible in vanilla to get her picture?

Like, she has a detection radius too large to even be able to get close enough, no?

Reply:
The capture check is for a cylinder of semiheight 100 and radius 700, so you can get closer than 400 if you're high enough up. Whether the window is within that I'm not sure. But you can certainly get a bad picture, I'd have thought.